test/e2e/hive_manager.go

```diff
@@ -0,0 +1,42 @@
+package e2e
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("Hive manager", func() {
+	Describe("CreateNamespace", func() {
+		var ns *corev1.Namespace
+		It("Should be created successfully", func() {
+			var err error
+			ns, err = clients.HiveClusterManager.CreateNamespace(context.Background())
+			Expect(err).NotTo(HaveOccurred())
+			Expect(ns).NotTo(BeNil())
+
+			res, err := clients.HiveAKS.CoreV1().Namespaces().Get(context.Background(), ns.Name, metav1.GetOptions{})
```

Comment:
would it make sense to have a timeout on that context so we can be sure that the test will finish in a specific amount of time ?

Reply:
Yeah, it makes sense to have time limits for tests. As I understand, there are some ways to set timeout via ginkgo – `SpecTimeout` and `NodeTimeout`. But these things are available only in the latest versions of ginkgo.

Now I am wondering why we don't have timeouts for k8s calls in all other places.

Yeah, we can set a timeout locally, using `context.WithTimeout`, but actually it is not the only option we have.
Also, we can set it once while initiating the k8s client in the e2e `setup()` function.

@m1kola could you share your thoughts, please?

Reply:
We limit e2e total test execution time to 180 minutes [here](https://github.com/Azure/ARO-RP/blob/4240ea46193a8a72d7275ecddcfb66406640e866/Makefile#L5). Ginkgo will fail the job if exeed 180 min.

> But these things are available only in the latest versions of ginkgo.

We already have [version of Ginkgo which supports this](https://github.com/Azure/ARO-RP/pull/2503), but I don't think that setting a custom timeout on a spec or node will be beneficial: global timeout seems to be enough for me. Or am I missing something?

But we must get rid of creating own context for each call and use context provided by a node/spec. I updated all of our existing tests to do that in [this PR](https://github.com/Azure/ARO-RP/pull/2505).

Reply:
@m1kola but if some test is stuck for some reason, we have to wait for ~180min to see it failed, right? does it make sense? It is actually the only case I wanna avoid by setting a timeout on node/spec.